### PR TITLE
[fix](match) fix match query with compound predicates return -6003

### PR DESCRIPTION
**Pull request by Tanya-W**
## Proposed changes

fix match query with compound predicates return -6003

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -717,10 +717,15 @@ Status SegmentIterator::_apply_index_except_leafnode_of_andnode() {
             continue;
         }
```
Comment (eldenmoon) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:718]: maybe we should add some test case?

**Comment by Tanya-W** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -717,10 +717,15 @@ Status SegmentIterator::_apply_index_except_leafnode_of_andnode() {
...
+                                             PredicateTypeTraits::is_equal_or_list(pred_type);
         if (!res.ok()) {
-            if ((res.code() == ErrorCode::INVERTED_INDEX_FILE_NOT_FOUND &&
-                 pred->type() != PredicateType::MATCH) ||
-                res.code() == ErrorCode::INVERTED_INDEX_FILE_HIT_LIMIT) {
+            if (res.code() == ErrorCode::INVERTED_INDEX_FILE_NOT_FOUND ||
+                res.code() == ErrorCode::INVERTED_INDEX_FILE_HIT_LIMIT ||
+                res.code() == ErrorCode::INVERTED_INDEX_EVALUATE_SKIPPED ||
+                (res.code() == ErrorCode::INVERTED_INDEX_NO_TERMS &&
+                 need_remaining_after_evaluate)) {
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:728]: wrap as a function, so _apply_inverted_index_on_column_predicate can also use it, and next time you change your condition, you will not forget that

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:718]: yes, I will add some regression case

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 2 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:728]: you are right, I will update it

**Comment by Tanya-W** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by Tanya-W** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**